packages/server/lib/server-base.ts

```diff
@@ -174,6 +167,39 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
     return this.ensureProp(this._httpsProxy, 'open')
   }
 
+  setupMultiDomain () {
+    this._eventBus.on('cross:domain:delaying:html', (request) => {
+      this.socket.localBus.once('ready:for:domain', () => {
+        this._eventBus.emit('ready:for:domain')
+      })
+
+      this.socket.localBus.once('ready:for:domain:failed', () => {
+        this._eventBus.emit('ready:for:domain:failed')
+      })
```

Comment:
We'll have the same issue with stacked up listeners here.

Reply:
Fixed [145ab0e](https://github.com/cypress-io/cypress/pull/20752/commits/145ab0ecf31f25e2ee69ba400fa5c662156019f7)